packages/contract-notifier/src/ContractNotifier.js

```diff
@@ -0,0 +1,119 @@
+   */
+  constructor({ logger, networker, getTime, apiEndpoint, maxTimeTillExpiration }) {
+    this.logger = logger;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeTillExpiration = maxTimeTillExpiration;
+  }
+
+  async checkUpcomingExpirations() {
+    this.logger.debug({
+      at: "ExpirationsNotifier",
+      message: "Checking for upcoming expirations",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeTillExpiration: this.maxTimeTillExpiration,
+    });
+
+    const currentTime = await this.getTime();
+    const notifiedExpirations = await this.getNotifiedExpirations();
+    const expirationPeriod =
+      this.maxTimeTillExpiration > 259200
+        ? parseInt(this.maxTimeTillExpiration / 86400) + " days"
+        : parseInt(this.maxTimeTillExpiration / 3600) + " hours";
+    const apiUrl = this.apiEndpoint + "/global/listActive";
+    const activeContracts = await this.networker.getJson(apiUrl, { method: "post" });
+    const expiringContracts = activeContracts
+      .map((contract) => {
+        if (!contract.type || !contract.expirationTimestamp) {
+          return null;
+        }
+        let tokenName;
+        if (contract.type === "emp") {
+          tokenName = contract.tokenName;
+        } else if (contract.type === "lsp") {
+          tokenName = contract.longTokenName;
+        } else {
+          tokenName = "";
+        }
+        const expirationUtcString = new Date(contract.expirationTimestamp * 1000).toUTCString();
+        return {
+          // UMA API currently supports only Ethereum mainnet, thus chainId of 1 is hardcoded here:
+          chainId: 1,
+          address: contract.address,
+          expirationTimestamp: contract.expirationTimestamp,
+          tokenName: tokenName,
+          expirationUtcString: expirationUtcString,
+        };
+      })
+      .filter((contract) => {
+        return (
+          contract &&
+          contract.expirationTimestamp - currentTime <= this.maxTimeTillExpiration &&
+          contract.expirationTimestamp > currentTime &&
+          (!Object.keys(notifiedExpirations).includes(contract.chainId + "_" + contract.address) ||
+            contract.expirationTimestamp -
+              notifiedExpirations[contract.chainId + "_" + contract.address].notificationTimestamp >
+              this.maxTimeTillExpiration)
+        );
+      });
+
+    let mrkdwn = `Following contracts are expiring in ${expirationPeriod}:`;
+    for (let contract of expiringContracts) {
+      mrkdwn =
+        mrkdwn +
+        `\n- ${createEtherscanLinkMarkdown(contract.address, contract.chainId)}:` +
+        ` ${contract.tokenName}` +
+        ` is expiring on ${contract.expirationUtcString}`;
+    }
+    if (expiringContracts.length) {
+      this.logger.warn({
+        at: "ExpirationsNotifier",
+        message: "Expiring contracts reminder 🔔",
+        mrkdwn,
+        notificationPath: "dev-x",
```

Comment:
good stuff! for this to work in prod we will need to update the bot config as well before this will actually send messages correctly.